tutorials/ollama_client_note.py

```diff
@@ -1,109 +1,46 @@
-    return client.list_objects_v2(Bucket=bucket)
-
-
-# Create components that will serve as function calls to our local LLM
-class S3Tools(Component):
-    def __init__(
-        self, model_client, model_kwargs, prompt_kwargs, output_processor, manager
-    ):
-        super().__init__()
-        self.generator = Generator(
-            model_client=model_client,
-            model_kwargs=model_kwargs,
-            prompt_kwargs=prompt_kwargs,
-            output_processors=output_processor,
-        )
-        self.manager = manager
-
-    # Override the call function to return the output of the function call.
-    def call(self, input: dict) -> str:
-        output: GeneratorOutput = self.generator(input)
-        func = Function.from_dict(output.data)
-        func_output = self.manager.execute_func(func)
-        return func_output.output
-
-
-# Create a component to synthesize the response.
-class Synth(Component):
-    def __init__(self, model_client, model_kwargs, prompt_kwargs):
-        super().__init__()
-        self.generator = Generator(
-            model_client=model_client,
-            model_kwargs=model_kwargs,
-            prompt_kwargs=prompt_kwargs,
-        )
-
-    def call(self, input: dict) -> str:
-        return self.generator.call({"input_str": str(input)})
-
-
+import asyncio
+
+
+# To get started with Ollama we're going to need to install it.
+# Navigate to https://ollama.com/download in order to get ollama installed locally.
+# Once you have ollama installed, we can pull the model we want to use with LightRAG.
+## `ollama pull qwen2:0.5b` will allow you to download the Qwen2 model ~ 352MB
+## **After installation, ollama should already be running in the background.**
+# See https://ollama.com/library/qwen2:0.5b
+# Once you've completed the steps above, you're ready to run the demo.
+
+# We'll define an async function ahead of time to illustrate how to use the asyncronous client
+async def poems_async(poem_generator: Generator, topics: list[str]) -> list[str]:
+    results = []
+    for topic in topics:
+        result = await poem_generator.acall({"topic": topic})
+        results.append(result.data)
+    return results
+
+# Let's show the clients in action
 if __name__ == "__main__":
-    model_kwargs = {"model": "internlm2"}
-    # Let's start with something simple, generate a poem
-    gen = Generator(model_client=OllamaClient(), model_kwargs=model_kwargs)
-    output = gen(
-        {
-            "input_str": "Generate a poem about the Utah sky, in the winter, after a midnight storm."
-        }
+    # We'll start with a poem
+    # Create the template for the Generator and define the model to use
+    poem_prompt_template = """Generate a haiku about {{topic}}."""
+    model_kwargs = {"model": "qwen2:0.5b"}
+    poem_generator = Generator(
+        model_client=OllamaClient(),
+        model_kwargs=model_kwargs,
+        template=poem_prompt_template,
     )
-    print(output.data)
```

Comment:
would be great to also add an embedder example

Reply:
Opened an issue with the embedder signature for acall not matching ModelClient acall signature.
https://github.com/SylphAI-Inc/LightRAG/issues/129